deps/rabbit/src/rabbit_channel.erl

```diff
@@ -2123,27 +2123,35 @@ deliver_to_queues({Delivery = #delivery{message    = Message = #basic_message{
                                         QRef = amqqueue:get_name(Q),
                                         [QRef | Acc]
                                 end, [], Qs),
-    {ok, QueueStates, Actions} =
-        rabbit_queue_type:deliver(Qs, Delivery, QueueStates0),
-    %% NB: the order here is important since basic.returns must be
-    %% sent before confirms.
-    ok = process_routing_mandatory(Mandatory, Qs, Message, State0),
-    State1 = process_routing_confirm(Confirm, AllQueueNames,
-                                     MsgSeqNo, XName, State0),
-    %% Actions must be processed after registering confirms as actions may
-    %% contain rejections of publishes
-    State = handle_queue_actions(Actions,
-                                 State1#ch{queue_states = QueueStates}),
-    case rabbit_event:stats_level(State, #ch.stats_timer) of
-        fine ->
-            ?INCR_STATS(exchange_stats, XName, 1, publish),
-            [?INCR_STATS(queue_exchange_stats,
-                         {amqqueue:get_name(Q), XName}, 1, publish)
-             || Q <- Qs];
-        _ ->
-            ok
-    end,
-    State.
+    try
```

Comment:
Is this try catch here just to catch init errors? Just wondering if queue_type:deliver can return an error instead to make it a bit more portable when we want to use the queue type outside of the channel.